controllers/chat/moveToAnon.js

```diff
@@ -0,0 +1,130 @@
+    }
+
+    const newChannel = client.channel('messaging', {members});
+    const createdChannel = await newChannel.create();
+
+    const targets = members.filter((member) => member !== req.userId);
+    await targets.map(async (target) => {
+      const userModel = await UsersFunction.findUserById(User, target);
+
+      if (userModel.is_anonymous) {
+        const checkChatAnonUserInfo = await ChatAnonUserInfo.findOne({
+          where: {
+            channel_id: oldChannelId,
+            my_anon_user_id: req.userId,
+            target_user_id: target
+          }
+        });
+
+        if (checkChatAnonUserInfo === null) {
+          await ChatAnonUserInfo.create({
+            channel_id: newChannel.id,
+            my_anon_user_id: req.userId,
+            target_user_id: target,
+            anon_user_info_color_code:
+              checkChatAnonUserInfo?.anon_user_info_color_code || anon_user_info_color_code,
+            anon_user_info_color_name:
+              checkChatAnonUserInfo?.anon_user_info_color_name || anon_user_info_color_name,
+            anon_user_info_emoji_code:
+              checkChatAnonUserInfo?.anon_user_info_emoji_code || anon_user_info_emoji_code,
+            anon_user_info_emoji_name:
+              checkChatAnonUserInfo?.anon_user_info_emoji_name || anon_user_info_emoji_name
+          });
+        }
+      }
+    });
+
+    try {
+      await newChannel.updatePartial({
+        set: {
+          channel_type: CHANNEL_TYPE.ANONYMOUS,
+          anon_user_info_color_code,
+          anon_user_info_color_name,
+          anon_user_info_emoji_code,
+          anon_user_info_emoji_name,
+          better_channel_member: newChannel.state.members
+        }
+      });
+    } catch (e) {
+      console.log(e);
+    }
+
+    const targetsUserModel = await UsersFunction.findMultipleUsersById(User, targets);
+    targetsUserModel.push({
+      user_id: req.userId,
+      username: `Anonymous ${anon_user_info_emoji_name}`,
+      profile_pic_path: '',
+      anon_user_info_color_code,
+      anon_user_info_color_name,
+      anon_user_info_emoji_code,
+      anon_user_info_emoji_name
+    });
+
+    // get 100 messages
+    const channelFilters = {cid: 'messaging:' + newChannel.id};
+    const messageFilters = {created_at: {$lte: new Date()}};
+    const messageHistory = await client.search(channelFilters, messageFilters, {
+      sort: [{updated_at: -1}],
+      limit: 100
+    });
+
+    await client.disconnectUser();
+
+    const response = {
+      channel: createdChannel,
+      members: targetsUserModel,
+      messageHistory: messageHistory.results
+    };
+
+    if (createdChannel?.channel?.is_channel_blocked) {
+      return res.status(400).json(responseError('Channel is blocked', response));
+    }
+
+    return res.status(200).json(responseSuccess('sent', response));
+  } catch (error) {
+    await client.disconnectUser();
+    return ErrorResponse.e400(res, error.message);
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the error handling block, you're disconnecting the user and then returning a 400 error with the error message. However, it might be more appropriate to return a 500 error here since this is an unexpected server error, not a client error.

```diff
-    return ErrorResponse.e400(res, error.message);
+    return ErrorResponse.e500(res, error.message);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->